libraries/chef_ingredient_resource.rb

```diff
@@ -21,22 +21,22 @@ class ChefIngredient < Chef::Resource::LWRPBase
 
       actions :install, :uninstall, :remove, :reconfigure, :upgrade
       default_action :install
-      state_attrs :installed
 
       attribute :product_name, kind_of: String, name_attribute: true
-      attribute :installed, kind_of: [TrueClass, FalseClass, NilClass], default: false
-      attribute :reconfigure, kind_of: [TrueClass, FalseClass], default: false
       attribute :config, kind_of: String, default: nil
 
+      # Attributes for determining what version to install from which channel
+      attribute :version, kind_of: [String, Symbol], default: :latest
+      attribute :channel, kind_of: Symbol, default: :stable, equal_to: [:current, :stable]
```

Comment:
IMO unless there is a reason we should not relax this in order not to sprinkle repo with `to_s`s. But if that is a good cookbook writing practice we can do that :smile: